core/server/common/src/main/java/alluxio/cli/validation/SecureHdfsValidationTask.java

```diff
@@ -58,6 +58,9 @@ public SecureHdfsValidationTask(String process) {
 
   @Override
   public boolean validate(Map<String, String> optionsMap) {
+    if (super.shouldSkip()) {
```

Comment:
what if we are using HDFS, but not secure HDFS?

Reply:
@aaudiber  +1. we can organize the validation tasks into two groups: required and optional. HDFS-related tasks should fall into the optional group.
@bf8086 What is your opinion? Thanks!

Reply:
I would prefer we run all tasks if they are applicable to the current configuration. The command should be smart about what needs validation instead of rely on user to decide what to run. There are already many validation tasks and it can be hard for user to realize what needs to be validated.

I think we should separate the HDFS check and security check so that we skip the security check if the HDFS is not secured.

Reply:
Tasks like `HdfsValidationTask` or `SecureHdfsValidationTask` are optional. When the user specifies target without task name, `bin/alluxio validateEnv local`, all tasks will run. In this case, these optional tasks should be able to find out whether they are applicable. For example, `HdfsValidationTask` must be able to detect whether the UFS is HDFS; `SecureHdfsValidationTask` must be able to check whether HDFS is using Kerberos.

Reply:
The straightforward, easy-to-implement solution is to make some checks non-default. The downside is that then the tool is less useful because users might not run the right checks.

The harder solution is to make the checks smarter so that they decide on their own whether they apply. This is better for users, but place too much burden and prevent us from writing useful checks because we can't tell whether they apply or not.

I propose implementing support for non-default checks, but striving to run as many checks as possible by default.

Reply:
We already have validation tasks skips either based on configuration or user input. For example, many tasks will simply skip if Alluxio is already running in the background; and HDFS configuration consistency check will skip by default unless user specifies where the server side hadoop configuration file is located. 

For determining whether we should skip secured HDFS validation, I suggest we just check whether the server principal is specified in the configuration, and skip if that is not set. We can print a message saying the validation is skipped for secured HDFS because the principal is not specified. This way we won't run this check unless the user has configured the secured HDFS. If they want to use secured HDFS, they will see the message and configure the server accordingly.

Reply:
This sounds good to me